### ISSUE TITLE
docker push with unexpected user prompt

### ISSUE THREAD

**Issue by xrow** [mitchellh/packer#2386]
Hi,

there seems to be a user promt "Do you really want to push to public registry?" when pushing an image. It looks like packer can`t handle the promt. Instread it answers wiht "EOF".

``` shell
[root@localhost centos-7.1.1503-x86_64]# packer -v
0.8.0
```

``` shell
[root@172-17-42-1 ~]# docker -v
Docker version 1.6.2, build ba1f6c3/1.6.2
```

``` shell
==> docker: Killing the container: 65708b2da76406be1cd95cdfafe7bb09d781ccb2000ff2361d81e67650b491bb
==> docker: Running post-processor: docker-import
    docker (docker-import): Importing image: Container
    docker (docker-import): Repository: xrow/centos7:0.1
    docker (docker-import): Imported ID: cbe1dfff1e5930b42cb4d1e95b27a3daa0fb48387c51bfd9be07e801c9dd4d1e
==> docker: Running post-processor: docker-push
    docker (docker-push): Pushing: xrow/centos7
    docker (docker-push): Do you really want to push to public registry? [y/n]: EOF
Build 'docker' errored: 1 error(s) occurred:

* Post-processor failed: Bad exit status: 1

==> Some builds didn't complete successfully and had errors:
--> docker: 1 error(s) occurred:

* Post-processor failed: Bad exit status: 1

==> Builds finished but no artifacts were created.
```

``` json
  "post-processors": [
        [
            {
                "type": "docker-import",
                "repository": "xrow/centos7",
                "tag": "0.1"
            },
            {
                "type": "docker-push",
                "login_username": "xrow",
                "login_password": "donottell"
            }
        ]
  ]
```

Maybe you can use this option "-f" and make it work.

``` shell
[root@localhost centos-7.1.1503-x86_64]# docker push --help

Usage: docker push [OPTIONS] NAME[:TAG]

Push an image or a repository to the registry

  -f, --force=false    Push to public registry without confirmation
```

**Comment by xrow** [comment]
Hopefully my configuration is correct. I hope one can confirm the error. My OS is CentOS 7.

**Comment by cbednarski** [comment]
It looks like `-f` was removed in docker `1.7.0`. Generally the way to handle this is to preemptively respond `y` to the prompt, but I'm not sure how easy it is for us to catch this retroactively.

**Comment by xrow** [comment]
I see and understand. The consequence is that without a fix. All rhel/centos 7 and earlier distros will work since the shipped version is docker 1.6.

You can use the expect http://linux.die.net/man/1/expect command to fix, but there must be a native implementation in the programming language you use.

**Comment by xrow** [comment]
Oh one Idea. What if you check the packer version upfront? If the you have a version lower 1.7.0 you add the -f.
